File: C++/memory.cpp
#include "header.hh"

/* Constructor. */

// Empty constructor.
Memory::Memory()
{

}

// Constructor with memory's data.
Memory::Memory(int frequency, int memorySize, bool supportsCuda)
{
    this->frequency = frequency;
    this->memorySize = memorySize;
    this->supportsCuda = supportsCuda;
}

/* Setter and Getter. */

// Set frequency attribute.
void Memory::setFrequency(int frequency)
{
    this->frequency = frequency;
}

// Set memory size attribute.
void Memory::setMemorySize(int memorySize)
{
    this->memorySize = memorySize;
}

// Set CUDA support attribute.
void Memory::setSupportsCuda(bool supportsCuda)
{
    this->supportsCuda = supportsCuda;
}

// Get frequency attribute.
int Memory::getFrequency()
{
    return frequency;
}

// Get memory size attribute.
int Memory::getMemorySize()
{
    return memorySize;
}

// Get CUDA support attribute.
bool Memory::getSupportsCuda()
{
    return supportsCuda;
}

/* Destructor. */

Memory::~Memory()
{
    
}

File: Python/Memory.py
from Hardware import Hardware

class Memory(Hardware):

    # Memory attributes.
    __frequency = 0
    __memorySize = 0
    __supportsCuda = False

    # Constructor.
    # Please take note that after a few try, Python seems to not accept
    # multiple constructor. So I decided to set it as empty constructor. #

    # Empty constructor.
    def __init__(self):
        pass
    
    # Setter and Getter. #

    # Set frequency attribute.
    def setFrequency(self, frequency):
        self.__frequency = frequency
    
    # Set memory size attribute.
    def setMemorySize(self, memorySize):
        self.__memorySize = memorySize
    
    # Set CUDA support attribute.
    def setSupportsCuda(self, supportsCuda):
        self.__supportsCuda = supportsCuda
    
    # Get frequency attribute.
    def getFrequency(self):
        return self.__frequency
    
    # Get memory size attribute.
    def getMemorySize(self):
        return self.__memorySize
    
    # Get CUDA support attribute.
    def getSupportsCuda(self):
        return self.__supportsCuda